Examples/Detectors/MuonSpectrometerMockupDetector/src/GeoModelMuonMockupBuilder.cpp

```diff
@@ -0,0 +1,144 @@
+    Acts::VolumeBoundFactory& boundFactory) const {
+  /** Assume a station paradigm. MDT multilayers and complementary strip
+   * detectors are residing under a common parent node representing a muon
+   * station envelope. Group the passed boxes under by their parent */
+  std::map<PVConstLink, GeoModelVolumeFPVsVec> commonStations{};
+  for (const auto& box : boundingBoxes) {
+    if (std::get<1>(box)->name().find(name) == std::string::npos) {
+      continue;  // skip boxes that do not match the station name
+    }
+    auto parent = std::get<2>(box)->getParent();
+
+    if (!parent) {
+      THROW_EXCEPTION("No parent found for " << name);
+    }
+    commonStations[parent].push_back(box);
+  }
+  // Create a vector to hold the chambers
+  std::vector<std::unique_ptr<Acts::TrackingVolume>> volChambers;
+  volChambers.reserve(commonStations.size());
+  std::size_t stationNum = 0;
+  for (const auto& [parentPhysVol, childrenTrkVols] : commonStations) {
+    std::shared_ptr<Acts::Volume> parentVolume = Acts::GeoModel::convertVolume(
+        parentPhysVol->getX(), parentPhysVol->getLogVol()->getShape(),
+        boundFactory);
+
+    std::unique_ptr<Acts::TrackingVolume> chamberVolume =
+        std::make_unique<Acts::TrackingVolume>(
+            *parentVolume, "Chamber_" + std::to_string(stationNum++));
+    chamberVolume->assignGeometryId(
+        Acts::GeometryIdentifier{}.withVolume(stationNum));
+
+    std::size_t childVol = 0;
+    for (const auto& child : childrenTrkVols) {
+      auto& vol = std::get<0>(child);
+
+      std::unique_ptr<Acts::TrackingVolume> trVol =
+          std::make_unique<Acts::TrackingVolume>(*vol,
+                                                 std::get<1>(child)->name());
+      trVol->assignGeometryId(Acts::GeometryIdentifier{}
+                                  .withVolume(stationNum)
+                                  .withLayer(childVol++));
+
+      // add the sensitives (tubes) in the constructed tracking volume
+      auto sensitives = std::get<1>(child)->surfacePtrs();
+
+      for (const auto& surface : sensitives) {
+        trVol->addSurface(surface);
+      }
+
+      chamberVolume->addVolume(std::move(trVol));
+    }
+    volChambers.push_back(std::move(chamberVolume));
+  }
+
+  const Acts::Vector3& cent{volChambers.front()->center()};
+  double rmincyl = Acts::fastHypot(cent.x(), cent.y()) -
+                   volChambers.front()->volumeBounds().values()[0];
+  double rmaxcyl = Acts::fastHypot(
+      rmincyl + 2 * volChambers.front()->volumeBounds().values()[0],
+      volChambers.front()->volumeBounds().values()[1]);
+
+  // Create the barrel node with the attached cylinder volume
+  auto barrelNode = std::make_shared<Acts::Experimental::StaticBlueprintNode>(
+      std::make_unique<Acts::TrackingVolume>(
+          Acts::Transform3::Identity(),
+          std::make_shared<Acts::CylinderVolumeBounds>(rmincyl, rmaxcyl, 4_m),
```

Comment:
So do I understand correctly that this is just creating a wrapping volume with the right radii to fit the chambers? I think this works, but ultimately this could possibly become a more general node type in the blueprint tree, that produces an auto-sized wrapping volume around it's children. Ok for now though.